sdk/monitor/azure-monitor-opentelemetry-autoconfigure/src/main/java/com/azure/monitor/opentelemetry/autoconfigure/implementation/LogDataMapper.java

```diff
@@ -86,14 +89,44 @@ public TelemetryItem map(LogRecordData log, @Nullable String stack, @Nullable Do
         if (sampleRate == null) {
             sampleRate = getSampleRate(log);
         }
-        if (stack == null) {
-            return createMessageTelemetryItem(log, sampleRate);
-        } else {
-            return createExceptionTelemetryItem(log, stack, sampleRate);
+
+        Attributes attributes = log.getAttributes();
+        String customEventName = attributes.get(AttributeKey.stringKey(CUSTOM_EVENT_NAME));
+        if (customEventName != null) {
+            return createEventTelemetryItem(log, attributes, customEventName, sampleRate);
         }
+
+        if (stack != null) {
+            return createExceptionTelemetryItem(log, attributes, stack, sampleRate);
+        }
```

Comment:
I think these two need to be swapped. Exception should take precedence over CustomEvents